xdsl/dialects/irdl/irdl.py

```diff
@@ -115,6 +153,24 @@ def qualified_name(self):
         return f"{dialect_op.sym_name.data}.{self.sym_name.data}"
 
 
+@irdl_op_definition
+class CPredOp(IRDLOperation):
+    """Constraints an attribute using a C++ predicate"""
+
+    name = "irdl.c_pred"
+
+    pred = attr_def(StringAttr)
+
+    output: OpResult = result_def(AttributeType())
+
+    assembly_format = "$pred attr-dict"
+
+    def __init__(self, pred: str | StringAttr):
+        if isinstance(pred, str):
+            pred = StringAttr(pred)
+        super().__init__(attributes={"pred": pred}, result_types=[AttributeType()])
+
```

Comment:
De we really reaaally want this one? I like my denial of its necessity :smiling_face_with_tear:

Reply:
Probably necessary if you ever want to read something made by the tblgen-to-irdl script